### PR TITLE
feat: support shape in edgeless mode

### PR DESCRIPTION
**Pull request by himself65**

### CODE REVIEW

**Review by himself65 — COMMENTED** [review]
`packages/blocks/src/shape-block/utils/vector.ts` is copied from `tldraw`

**Comment by himself65** [comment]
image

```ts
<svg style=${styleMap({top: '100px', left: '100px', position: 'absolute'})}>
  <path d=${getRectanglePath({}, [100, 100])} stroke="2" />
</svg>
```

**Comment by himself65** [comment]
image

**Comment by tzhangchi** [comment]
Great job

**Comment by himself65** [comment]
I'm working on the logic about darg/drop/resize/move logic of shape block

**Comment by himself65** [comment]
The basic usage of shape has been stable. Just need to fix the bugs on dragging, resizing and test cases.

File: packages/blocks/src/shape-block/shape-block.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      `;
+    }
+    console.error('not supported');
+    return html``;
+  }
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'pencil-block': ShapeBlockComponent;
```
Comment (doodlewind) [inline packages/blocks/src/shape-block/shape-block.ts:47]: Should be `shape-block`?

File: packages/blocks/src/shape-block/shape-model.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { BaseBlockModel, IBaseBlockProps, Page } from '@blocksuite/store';
+import type { ShapeColor, ShapeType } from '../__internal__';
+
+export interface ShapeBlockProps extends IBaseBlockProps {
+  flavor: 'affine:shape';
+  color: ShapeColor;
+  type: ShapeType;
+
+  // Refs: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d
+  d: string;
```
Comment (doodlewind) [inline packages/blocks/src/shape-block/shape-model.ts:10]: Should we add the `d` and `color` fields to the `ShapeBlockModel`?

File: packages/blocks/src/shape-block/utils/vector.ts [hunk 3]
```diff
@@ -0,0 +1,585 @@
+export class Vec {
```
Comment (doodlewind) [inline packages/blocks/src/shape-block/utils/vector.ts:1]: Do we need to add comments about the credit of the original module?

File: packages/blocks/src/__internal__/utils/types.ts [hunk 4]
```diff
@@ -38,9 +40,17 @@ export interface BlockSelectionInfo {
   blocks: SelectedBlock[];
 }
 
+// blocks that would only appear under the root
+export type RootBlockModels = GroupBlockModel | ShapeBlockModel
```
Comment (doodlewind) [inline packages/blocks/src/__internal__/utils/types.ts:44]: How about `RootBlockModels` -> `RootBlockModel`?

**Review by doodlewind — APPROVED** [review]
Great start, let's keep the pace!

File: packages/blocks/src/shape-block/shape-model.ts (hunk 2 above)
Comment (himself65) [reply packages/blocks/src/shape-block/shape-model.ts]: Color field haven't been used but must be used in the future.

**Pull request merged**